misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt

```diff
@@ -131,26 +131,13 @@ internal class WebActionsServlet @Inject constructor(
 
       val httpCall = ServletHttpCall.create(
         request = request,
-        linkLayerLocalAddress = with((request as? Request)?.httpChannel) {
-          when (this?.connector) {
-            is UnixDomainServerConnector -> SocketAddress.Unix(
-              (this.connector as UnixDomainServerConnector).unixDomainPath.toString()
-            )
-
-            is UnixSocketConnector -> SocketAddress.Unix(
-              (this.connector as UnixSocketConnector).unixSocket
-            )
-
-            is ServerConnector -> SocketAddress.Network(
-              this.endPoint.remoteAddress.address.hostAddress,
-              (this.connector as ServerConnector).localPort
-            )
-
-            else -> throw IllegalStateException("Unknown socket connector.")
-          }
-        },
+        linkLayerLocalAddress = extractLinkLayerLocalAddress(request),
         dispatchMechanism = dispatchMechanism,
-        upstreamResponse = JettyServletUpstreamResponse(response as Response),
+        upstreamResponse = if (response is Response) {
```

Comment:
So the Jetty Servelet can technically have non-Jetty typed responses and that's what will allow this?